### ISSUE TITLE
[BUG] KubeConfig generated by Rancher uses wrong certificate-authority-data

### ISSUE THREAD

**Issue by braninl** [rancher/rancher#43309]
**Rancher Server Setup**
- Rancher version:  v2.7.6
- Installation option:  Helm Chart
   - k3s 1.26.9+k3s1
- Proxy/Cert Details:  See bug notes below

**Information about the Cluster**
- Kubernetes version:  1.26.9+k3s1
- Cluster Type (Local/Downstream):  Local (but custom clusters built from Rancher exhibit the same issue)

**User Information**
- What is the role of the user logged in?  Admin

**Describe the bug**
A 3 node k3s cluster was created and cert-manager was installed.  A ClusterIssuer was created for Let's Encrypt production using the DNS-01 solver (since our cluster is on-premise without external access).  A certificate for rancher.domain.com was requested and successfully generated and saved in the tls-rancher-ingress secret in the cattle-system namespace.  Rancher was then installed via Helm with the hostname set to rancher.domain.com and the ingress.tls.source set to secret.  Rancher successfully installed and I am able to access Rancher via https://rancher.domain.com with a production Let's Encrypt certificate and no errors.  I was able to also successfully create a k3s cluster through Rancher on another set of nodes without any issues.

However, within Rancher, when I click on the Download KubeConfig button (or the "Copy KubeConfig to Clipboard" button, the KubeConfig that is generated includes the wrong certificate-authority-data.  Specifically, after decoding the data in the KubeConfig file, it appears the certificate is coming from either the automatically created tls-rancher or tls-rancher-internal-ca secrets (as the domain name is dynamiclistener-ca@xxxxx with a matching expiration date to the secret).  This happens for both "Download KubeConfig" in the local cluster and a downstream cluster.

Attempting to kubectl get nodes with this KubeConfig file generates a "certificate signed by unknown authority" error.  Using --insecure-skip-tls-verify allows kubectl to work, which makes sense (but certainly isn't ideal).  Manually modifying the KubeConfig file to replace the "certificate-authority-data" with the correct CA data (from the tls-rancher-ingress secret) also allows kubectl to work, but the Download KubeConfig button in Rancher should just use the correct certificate-authority-data in the first place.

**To Reproduce**
1) Confirm the Rancher web UI is working correctly with a production Let's Encrypt generated certificate (without TLS errors), which was generated through Cert-Manager (installed before Rancher) and saved as tls-rancher-ingress in the cattle-system namespace, with Rancher installed with the correct hostname and ingress.tls.source set to secret.
2) Click Download KubeConfig in the Rancher UI.

**Result**
Kubectl complains about certificate issues with the downloaded KubeConfig, since it's using the wrong CA information.

**Expected Result**
"Download KubeConfig" should use the CA information from the tls-rancher-ingress secret.

**Screenshots**
tls-ran secrets in the cattle-system namespace
![image](https://github.com/rancher/rancher/assets/66883280/59ef6c01-bfdf-4a00-be86-69bf42383b69)

CA information in "Download KubeConfig" config file decoded
![image](https://github.com/rancher/rancher/assets/66883280/5d1aecac-47f0-44ce-a2bf-b140cf2335c0)

CA information in proper CA from actual certificate decoded
![image](https://github.com/rancher/rancher/assets/66883280/ceb95074-d747-4fdc-ada0-a20579bd494f)

**Comment by braninl** [comment]
For anyone that finds this in the future, this is not a bug after all.  I needed to create the tls-ca secret and add privateCA: true to the Rancher Helm Chart values to get this to work correctly.  I neglected doing that initially because Let's Encrypt isn't a private CA.  However, it's now obvious that Rancher "thinks" that it was a private CA, since I told Rancher to just use the secret.  Everything is working correctly now.

**Comment by pensionars** [comment]
Do I need to somehow generate a CA secret?

I am using a secret generated by cert-manager and letsencrypt. I set my values.yaml to this and updated my local cluster. Sadly, the UI still gives me the old certificate. Currently as a workaround I have saved the certificate to tls.crt and added it to my kube config but I would like to know how to do it correctly.

```
replicas: 1
ingress.tls.source: secret
privateCA: true 
```
